src/manifest.rs

```diff
@@ -0,0 +1,30 @@
+use anyhow::anyhow;
+use std::path::{Path, PathBuf};
+
+use crate::opts::DEFAULT_MANIFEST_FILE;
+
+pub(crate) fn manifest_path(path: impl AsRef<Path>) -> anyhow::Result<PathBuf> {
+    let path = path.as_ref();
+    if path.exists() {
```

Comment:
Consider [`try_exists`](https://doc.rust-lang.org/std/fs/fn.try_exists.html) which returns `Result(bool)`, erroring if existence cannot be determined due to e.g. permissions issues. Could flatten that with the nested conditionals since `is_file` and `is_dir` both imply `exists`.